src/hotspot/share/gc/parallel/psParallelCompact.cpp

```diff
@@ -1161,17 +1100,19 @@ bool PSParallelCompact::invoke_no_policy(bool clear_all_soft_refs) {
     heap->update_counters();
 
     heap->post_full_gc_dump(&_gc_timer);
+
+    size_policy->record_gc_pause_end_instant();
```

Comment:
What about moving this invocation into `major_collection_end`? Just like the `record_gc_pause_start_instant` and `major_collection_begin`.